gallery/visualtests/nxMultiFileUpload.js

```diff
@@ -0,0 +1,545 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+const fs = require('fs');
+const path = require('path');
+const tmp = require('tmp-promise');
+
+const { setupBrowser } = require('./testUtils');
+
+async function fillFile(path, numBytes) {
+  const MAX_BUFFER_SIZE = 1 << 20, // 1 MiB
+      writeStream = fs.createWriteStream(path);
+
+  let buffer;
+
+  for (let i = 0; i < numBytes; i += MAX_BUFFER_SIZE) {
+    const bufferSize = Math.min(MAX_BUFFER_SIZE, numBytes - i);
+
+    if (!(buffer && buffer.length === bufferSize)) {
+      buffer = await Buffer.alloc(bufferSize);
+    }
+
+    writeStream.write(buffer);
+  }
+
+  return new Promise((resolve, reject) => {
+    writeStream.on('finish', resolve);
+    writeStream.on('error', reject);
+
+    writeStream.end();
+  });
+}
+
+describe('NxMultiFileUpload', function() {
+  const {
+    checkScreenshot,
+    isInDocument,
+    waitAndGetElements,
+    isFocused,
+    clickTest,
+    focusTest,
+    focusAndHoverTest,
+    getPage,
+    hoverTest,
+    simpleTest,
+    a11yTest
+  } = setupBrowser('#/pages/Multiple File Upload');
+
+  const complexExampleSelector = '#nx-multi-file-upload-complex-example .gallery-example-live',
+      btnSelector = `${complexExampleSelector} .nx-multi-file-upload__select-btn`;
+
+  // Files of varying sizes to test the file size display in the component.  Due to the size
+  // of the largest of these files, there are problems with storing them in git so they
+  // are generated on the fly in beforeAll.
+  let files, tmpDir;
+
+  beforeAll(async function() {
+    tmpDir = await tmp.dir({ unsafeCleanup: true });
+
+    files = {
+      bytes: path.join(tmpDir.path, 'bytes'),
+      kilobytes: path.join(tmpDir.path, 'kilobytes'),
+      megabytes: path.join(tmpDir.path, 'megabytes'),
+      gigabytes: path.join(tmpDir.path, 'gigabytes-gigalongname')
+    };
+
+    const bytesPromise = fillFile(files.bytes, 14),
+        kilobytesPromise = fillFile(files.kilobytes, 2000),
+        megabytesPromise = fillFile(files.megabytes, 1500100),
+        gigabytesPromise = fillFile(files.gigabytes, 1200000100);
+
+    await Promise.all([bytesPromise, kilobytesPromise, megabytesPromise, gigabytesPromise]);
+  });
+
+  afterAll(async function() {
+    await tmpDir.cleanup();
+  });
+
```

Comment:
All of this can probably also be extracted out into a helper in `testUtils`

Reply:
> All of this can probably also be extracted out into a helper in `testUtils`

Moved into `testUtils` here: [9af2ab1](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/9af2ab1f211abf5599fffcf9e1624c07875d7e75). I created 2 functions, one to deal with the `beforeAll` hook, and one for cleanup in the `afterAll` hook. If there is a better way, then just let me know and I can adjust.